core/src/progress_map.rs

```diff
@@ -62,23 +64,39 @@ impl ReplaySlotStats {
             ),
             ("total_entries", num_entries as i64, i64),
             ("total_shreds", num_shreds as i64, i64),
-            ("check_us", self.execute_timings.check_us, i64),
-            ("load_us", self.execute_timings.load_us, i64),
-            ("execute_us", self.execute_timings.execute_us, i64),
-            ("store_us", self.execute_timings.store_us, i64),
+            (
+                "check_us",
+                self.execute_timings.metrics[ExecuteTimingType::CheckUs],
```

Comment:
These direct indexes worry me to, safe here, yes, safe in the future, ?

Reply:
That's a good point, thanks for pointing this out! I tried wrapping the array in a newtype prevent users from calling `timings.metrics[<bad_usize>]`. Let me know if there's a better alternative.